### PR TITLE
Npm/config

### PR DESCRIPTION
**Pull request by hilmarf**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue #633

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mandelsoft — CHANGES_REQUESTED** [review]

File: pkg/contexts/credentials/repositories/npm/provider.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	for key, value := range all {
+		u, e := url.Parse(key)
+		if e != nil {
+			return nil, nil
+		}
+
+		attrs := []string{identity.ID_HOSTNAME, u.Hostname()}
+		if u.Port() != "" {
+			attrs = append(attrs, identity.ID_PORT, u.Port())
+		}
```
[the remark this answers is not in the document]
Comment (hilmarf) [reply pkg/contexts/credentials/repositories/npm/provider.go:49]: correct, but where do I get it from?

File: pkg/contexts/credentials/repositories/npm/config.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// RepositorySpec describes a docker npmrc based credential repository interface.
+type RepositorySpec struct {
+	runtime.ObjectVersionedType `json:",inline"`
+	NpmrcFile                   string `json:"npmrcFile,omitempty"`
+}
+
+// NewRepositorySpec creates a new memory RepositorySpec.
+func NewRepositorySpec(path string, prop ...bool) *RepositorySpec {
+	p := false
+	for _, e := range prop {
```
[the remark this answers is not in the document]
Comment (hilmarf) [reply pkg/contexts/credentials/repositories/npm/config.go:49]: obsolete... useless leftover from copy&paste - will delete

File: pkg/contexts/credentials/repositories/npm/repository.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	if err != nil {
+		return errors.Wrapf(err, "cannot resolve path %q", r.path)
+	}
+	cfg, err := readNpmConfigFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to load npmrc: %w", err)
+	}
+	id := cpi.ProviderIdentity(PROVIDER + "/" + path)
+
+	r.ctx.RegisterConsumerProvider(id, &ConsumerProvider{path})
```
[the remark this answers is not in the document]
Comment (hilmarf) [reply pkg/contexts/credentials/repositories/npm/repository.go:73]: but I want this to always register...

[the remark this answers is not in the document]
Comment (mandelsoft) [reply pkg/contexts/credentials/repositories/npm/repository.go:73]: It should be a consistent behaviour for all externsions.

**Review by mandelsoft — APPROVED** [review]

**Pull request merged**